src/libxtp/qmpackages/orca.cc

```diff
@@ -825,12 +825,38 @@ std::string Orca::WriteMethod() const {
   std::string convergence =
       this->_convergence_map.at(_settings.get("convergence_tightness")) +
       "SCF ";
-  stream << "! DFT " << _settings.get("functional") << " " << convergence
+  stream << "! DFT " << this->GetOrcaFunctionalName() << " " << convergence
          << opt
          // additional properties provided by the user
          << _settings.get("orca.method") << "\n";
   return stream.str();
 }
 
+std::string Orca::GetOrcaFunctionalName() const {
+
+  char* votca_share = getenv("VOTCASHARE");
+  if (votca_share == nullptr) {
+    return _settings.get("functional");
+  } else {
+    tools::Property all_functionals;
+
+    auto xml_file = std::string(getenv("VOTCASHARE")) +
+                    std::string("/xtp/packages/orca_functional_names.xml");
+
+    all_functionals.LoadFromXML(xml_file);
+
+    const tools::Property& functional_names =
+        all_functionals.get("functionals");
+
+    std::string input_name = _settings.get("functional");
+
+    if (functional_names.exists(input_name)) {
+      return functional_names.get(input_name).as<std::string>();
+    } else {
+      return input_name;
```

Comment:
should we not throw an error here?

Reply:
I am not sure if that is the best solution. We will throw an error based on the assumption that *Orca* doesn't know the functional that the user has passed based on an small subset of functional that we have. But I suppose that for most of the users an error in this point is more meaningful that in the Orca runtime.
I am going to change it.